### PR TITLE
CLOUDP-70049: mongocli atlas security ldap verify status watch <ID>

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-70049

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
./bin/mongocli atlas security ldap verify status watch 5f5b3172dc814849f90daa22

LDAP Configuration request completed.
```

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
@gssbzn tagging you here so you can review this PR once you're back.

File: internal/cli/atlas/security/ldap/status.go [hunk 1]
```diff
@@ -49,13 +49,13 @@ func (opts *StatusOpts) Run() error {
...
 
-// mongocli atlas ldap status <ID> [--projectId projectId]
+// mongocli atlas ldap verify status <ID> [--projectId projectId]
 func StatusBuilder() *cobra.Command {
 	opts := &StatusOpts{}
 	cmd := &cobra.Command{
 		Use:   "status <ID>",
 		Args:  cobra.ExactValidArgs(1),
-		Short: verify,
+		Short: status,
```
Comment (colm-quinn) [inline internal/cli/atlas/security/ldap/status.go:58]: [q] Is `verify` removed, as I see there's still a definition in `description.go`?

Comment (andreaangiolillo) [reply internal/cli/atlas/security/ldap/status.go:58]: no, I forgot to change the description of the status command in the previous PR. The verify command is still there.

File: internal/cli/atlas/security/ldap/status.go [hunk 2]
```diff
@@ -70,5 +70,9 @@ func StatusBuilder() *cobra.Command {
 	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
 	cmd.Flags().StringVarP(&opts.Output, flag.Output, flag.OutputShort, "", usage.FormatOut)
 
+	cmd.AddCommand(
+		WatchBuilder(),
+	)
+
```
Comment (colm-quinn) [inline internal/cli/atlas/security/ldap/status.go:76]: [q] Why does the `StatusBuilder` then invoke the `WatchBuilder`? Is this a pattern used in other parts of the cli, as it's a bit confusing to read here.

**Review by colm-quinn — COMMENTED** [review]
Code looks good, just a few questions on how the changes are designed as I haven;t looked at this code much before.

File: internal/cli/atlas/security/ldap/status.go (hunk 2 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/security/ldap/status.go:76]: yes, I used the same pattern in `VerifyBuilder`, which calls the `StatusBuilder`. I am not sure if it is used in other part of project though but this is the only way that I found to define `atlas security ldap verify status watch <ID>`, indeed the user has to type `status` to use `watch`.
Happy to discuss this with @gssbzn as well to find a better way to do it

Comment (colm-quinn) [reply internal/cli/atlas/security/ldap/status.go:76]: Right, the commands are chaining off each other, ok that makes more sense.

**Review by colm-quinn — APPROVED** [review]
LGTM, may want to have a chat with @gssbzn when he's back about the Builders, but not blocking a merge now.

**Pull request merged**